lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3971,10 +3971,25 @@ class ConvertAtenPoolingBaseOp : public OpConversionPattern<AtenOpT> {
       return rewriter.notifyMatchFailure(
           op, "Failed to process inputs for pooling");
 
-    auto pooledOutput =
-        rewriter
-            .create<TosaOpT>(op->getLoc(), outputTy, input, kernel, stride, pad)
-            .getResult();
+    Value pooledOutput;
+    static_assert(std::is_same<TosaOpT, tosa::MaxPool2dOp>::value ||
+                      std::is_same<TosaOpT, tosa::AvgPool2dOp>::value,
+                  "Expected either tosa::MaxPool2dOp or tosa::AvgPool2dOp");
```

Comment:
There's probably some slicker way to "spell" this so feel free to drop a hint (or link) if your template-fu is better than mine.